packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol

```diff
@@ -395,67 +342,96 @@ contract RoutersFacet is BaseConnextFacet {
+  function setRouterRecipient(address _router, address _recipient) external onlyRouterOwner(_router) {
+    _setRouterRecipient(_router, _recipient, s.routerConfigs[_router].recipient);
   }
 
   /**
    * @notice Current owner or router may propose a new router owner
-   * @param router Router address to set recipient
-   * @param proposed Proposed owner Address to set to router
+   * @dev If routers burn their ownership, they can no longer update the recipient
+   * @param _router Router address to set recipient
+   * @param _proposed Proposed owner Address to set to router
    */
-  function proposeRouterOwner(address router, address proposed) external onlyRouterOwner(router) {
+  function proposeRouterOwner(address _router, address _proposed) external onlyRouterOwner(_router) {
+    // NOTE: If routers burn their ownership, they can no longer update the recipient
+
     // Check that proposed is different than current owner
-    if (getRouterOwner(router) == proposed) revert RoutersFacet__proposeRouterOwner_notNewOwner();
+    RouterConfig memory config = s.routerConfigs[_router];
+    if (config.owner == _proposed) revert RoutersFacet__proposeRouterOwner_notNewOwner();
 
     // Check that proposed is different than current proposed
-    address _currentProposed = s.routerPermissionInfo.proposedRouterOwners[router];
-    if (_currentProposed == proposed) revert RoutersFacet__proposeRouterOwner_badRouter();
+    if (config.proposed == _proposed) revert RoutersFacet__proposeRouterOwner_badRouter();
 
     // Set proposed owner + timestamp
-    s.routerPermissionInfo.proposedRouterOwners[router] = proposed;
-    s.routerPermissionInfo.proposedRouterTimestamp[router] = block.timestamp;
+    s.routerConfigs[_router].proposed = _proposed;
+    s.routerConfigs[_router].proposedTimestamp = block.timestamp;
 
     // Emit event
-    emit RouterOwnerProposed(router, _currentProposed, proposed);
+    emit RouterOwnerProposed(_router, config.proposed, _proposed);
   }
 
   /**
    * @notice New router owner must accept role, or previous if proposed is 0x0
-   * @param router Router address to set recipient
+   * @param _router Router address to set recipient
    */
-  function acceptProposedRouterOwner(address router) external onlyProposedRouterOwner(router) {
-    address owner = getRouterOwner(router);
+  function acceptProposedRouterOwner(address _router) external {
+    RouterConfig memory config = s.routerConfigs[_router];
 
     // Check timestamp has passed
-    if (block.timestamp - s.routerPermissionInfo.proposedRouterTimestamp[router] <= _delay)
+    if (block.timestamp - config.proposedTimestamp <= _delay)
       revert RoutersFacet__acceptProposedRouterOwner_notElapsed();
 
-    // Get current owner + proposed
-    address _proposed = s.routerPermissionInfo.proposedRouterOwners[router];
+    // Check the caller
+    address expected = config.proposed == address(0) ? config.owner : config.proposed;
```

Comment:
So it seems that setting the owner to address(0) will effectively prevent the router owner from changing the recipient again in the future. If someone does this by mistake, and they have the recipient set to an undesirable address (e.g an EOA with a typo) they can potentially brick their router, as they will be unable to withdraw their funds to an EOA they control.

This doesn't need to get fixed, but I'm just emphasizing it here as one of the pitfalls of running a router worth documenting. We need to be sure router operators understand the implications of setting owner to address(0).

Reply:
yeah, added a comment in propose for this